paymentsheet/src/main/java/com/stripe/android/paymentsheet/Element.kt

```diff
@@ -87,43 +72,59 @@ internal fun List<FormElement>.getIdInputControllerMap() = this
         this
             .filterIsInstance<FormElement.SectionElement>()
             .flatMap { it.fields }
-            .associate { it.identifier to it.controller }
+            .filter { it.controller is InputController }
+            .associate { it.identifier to it.controller as InputController }
     )
 
 /**
  * This is an element that is in a section and accepts user input.
  */
 internal sealed class SectionFieldElement {
     abstract val identifier: IdentifierSpec
-    abstract val controller: InputController
 
-    data class Name(
-        override val identifier: IdentifierSpec,
-        override val controller: TextFieldController
-    ) : SectionFieldElement(), SectionFieldElementType.TextFieldElement
+    /**
+     * Every item in a section must have a controller that can provide an error
+     * message, for the section controller to reduce it to a single error message.
+     */
+    abstract val controller: SectionFieldErrorController
+
+    /**
+     * This will return a controller that abides by the SectionFieldErrorController interface.
+     */
+    abstract fun sectionFieldErrorController(): SectionFieldErrorController
```

Comment:
Since `sectionFieldErrorController` is always the `controller`, can that just be defined here and the data classes don't need to override it?
```
fun sectionFieldErrorController(): SectionFieldErrorController = controller
```

Reply:
👍